tool/site/client-react/src/components/prescription-summary/medicationSummary.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from "react"
+import {MedicationRequest} from "fhir/r4"
+import {Table} from "nhsuk-react-components"
+
+export function createSummaryMedication(medicationRequest: MedicationRequest): SummaryMedication {
+  const quantity = medicationRequest.dispenseRequest.quantity
+  const snomedInformation = medicationRequest.medicationCodeableConcept.coding[0]
+
+  const summary: SummaryMedication = {
+    prescriptionEndorsements: [],
+    quantityUnit: quantity.unit,
+    quantityValue: quantity.value,
+    snomedCode: snomedInformation.code,
+    snomedCodeDescription: snomedInformation.display
+  }
+
+  if (medicationRequest.note) {
+    summary.dispenserNotes = medicationRequest.note
+      .filter(note => note.text)
+      .map(note => note.text)
+  }
+
+  if (medicationRequest.dosageInstruction) {
+    summary.dosageInstruction = medicationRequest.dosageInstruction
+      .filter(dosage => dosage.text)
+      .map(dosage => dosage.text)
+  }
+
+  return summary
+}
+
+export interface SummaryMedication {
+  dispenserNotes?: Array<string>
+  dosageInstruction?: Array<string>
+  prescriptionEndorsements: Array<string>
+  quantityUnit: string
+  quantityValue: number
+  snomedCode: string
+  snomedCodeDescription: string
+}
+
+interface MedicationSummaryProps {
+  medicationSummaryList: Array<SummaryMedication>
+}
+
+const MedicationSummary: React.FC<MedicationSummaryProps> = ({medicationSummaryList}) => {
+  return (
+    <Table.Panel heading="Medication">
+      <Table caption="Medication summary">
+        <Table.Head>
+          <Table.Row>
+            <Table.Cell>Details</Table.Cell>
+            <Table.Cell>Quantity</Table.Cell>
+            <Table.Cell>Unit</Table.Cell>
+            <Table.Cell>Dosage Instructions</Table.Cell>
+          </Table.Row>
+        </Table.Head>
+        <Table.Body>
+          {medicationSummaryList.map(medication => <MedicationRow {...medication} />)}
```

Comment:
I think the `MedicationRow` will need a `key` property or React will log a bunch of warnings to the console. You can just use the array index.